### DISCUSSION TITLE
1-wire temperature sensors

### DISCUSSION THREAD

**Discussion by pmitham** [hatlabs/discussions#92] [Q&A]
There is a header for 1-wire devices.  I've attached a  DS18B20  temp sensor to the board. Normally you need to bridge the 3v3 and DQ with a resistor.  Here there is no mention of it  and my temp sensor  (works fine on other boards)  is reporting a temp of 299c  what am I missing?

**Comment by mairas** [comment]
I'm assuming this question refers to HALMET. (Although, the 1-Wire design in both SH-ESP32 and HALMET are identical, so I guess it doesn't really matter...)

The 1-Wire connector already includes the pull-up resistor and some protection elements, so it should work as is.

The HALMET documentation is not yet finished but for 1-Wire information, you can refer to the SH-ESP32 documentation since the implementation is identical. Have a look e.g. at this tutorial: https://docs.hatlabs.fi/sh-esp32/pages/tutorials/onewire-temperature/
